R/zchunk_LB113.bio_Yield_R_GLU.R

```diff
@@ -25,53 +29,77 @@ module_aglu_LB113.bio_Yield_R_GLU_DISABLED <- function(command, ...) {
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Calculate global average yields for each FAO crop in the base year (31-39 old file)
+    L100.LDS_ag_HA_ha %>%
+      group_by(GTAP_crop) %>%
+      summarise(HA_ha = sum(value)) ->
+      L113.ag_HA_ha_glbl_crop
+
+    L100.LDS_ag_prod_t %>%
+      group_by(GTAP_crop) %>%
+      summarise(value = sum(value)) %>%
+      left_join_error_no_match(L113.ag_HA_ha_glbl_crop, by = "GTAP_crop") %>%
+      mutate(Yield_avg = value / HA_ha) ->
+      L113.ag_prod_t_glbl_crop
+
+    # Calculate each region / zone / crop's comparative yield (41-50)
+    L100.LDS_ag_HA_ha %>%
+      rename(HA = value) %>%
+      left_join_error_no_match(L100.LDS_ag_prod_t, by = c("iso", "GLU", "GTAP_crop")) %>%
+      mutate(Yield = value / HA) %>%
+      # Drop the missing values, where the harvested area was above the min threshold but production was not
+      na.omit ->
+      LDS_ag_Yield_tha
+
+    # Match in the global avg yield for each crop, and compute the ratio from that yield (52-61)
+    LDS_ag_Yield_tha %>%
+      left_join_error_no_match(select(L113.ag_prod_t_glbl_crop, GTAP_crop, Yield_avg), by = "GTAP_crop") %>%
+      mutate(Ratio = Yield / Yield_avg,
+             Ratio_weight = Ratio * HA) %>%
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      group_by(GCAM_region_ID, GLU) %>%
+      summarise(HA = sum(HA), Ratio_weight = sum(Ratio_weight)) %>%
+      ungroup %>%
+      mutate(YieldIndex = Ratio_weight / HA) ->
+      L113.YieldIndex_R_GLU
+
+    # Bioenergy yields are equal to this region/zone-specific index multiplied by a base yield
+    # The base yield is taken to be the maximum of the yields in the USA region, or the region
+    # containing the USA, because the Wullschleger paper (10.2134/agronj2010.0087)
+    # from which the yield estimate was derived was for the USA.
+    iso_GCAM_regID %>%
+      filter(iso == "usa") %>%
+      .[["GCAM_region_ID"]] ->
+      USAreg
+    L113.base_bio_yield_tha <- aglu.MAX_BIO_YIELD_THA / max(L113.YieldIndex_R_GLU$YieldIndex[L113.YieldIndex_R_GLU$GCAM_region_ID == USAreg])
```

Comment:
I don't have a strong preference. I guess I'd say this style, because dplyr 0.7 apparently has a new `pull` function for this, and it'll be easy to replace the `.[[` construct then.